docs/controllers/leaderelection.md

```diff
@@ -0,0 +1,62 @@
+# Leader Election
+
+This chapter is a stub about Leases, Leader Election and how to achieve distributed locks around Kubernetes resources.
+
+!!! note "Terminology"
+
+    See [Kubernetes//Leases](https://kubernetes.io/docs/concepts/architecture/leases/) for introductory definitions.
+
+
+## Crates
+
+At the moment, leader election support is not exported from `kube` direct, and requires 3rd party crates (as decided in [kube#485](https://github.com/kube-rs/kube/issues/485#issuecomment-1837386565)).
+
+### kube-leader-election
+The [`kube-leader-election` crate](https://crates.io/crates/kube-leader-election/) via [hendrikmaus/kube-leader-election](https://github.com/hendrikmaus/kube-leader-election) implements a simple and low-level form of leader election with a [use-case disclaimer](https://github.com/hendrikmaus/kube-leader-election?tab=readme-ov-file#kubernetes-lease-locking).
```

Comment:
Honestly, this disclaimer should be required reading, and is going to apply _to some extent_ to any attempt to build leader election on top of k8s.

Reply:
i'll add a section to the top on of this doc with a disclaimer and purpose. i think there's plenty of good arguments for not adopting a system like this (1 replica is fine unless you have crazy uptime requirements or load - which is often not the case in a controller), so maybe that should be highlighted rather than "here are the ways".

Reply:
have written a larger preamble about this now. It's [rendered here](https://github.com/kube-rs/website/blob/leader-election-stub/docs/controllers/leaderelection.md#requirements).

Reply:
think i have removed all value-laden adjectives now